runtime-decompiler/src/main/java/org/jrd/backend/core/DecompilerRequestReceiver.java

```diff
@@ -103,7 +103,7 @@ private int getPort(String hostname, int listenPort, String vmId, int vmPid) {
             actualListenPort = listenPort;
         }
         if (actualListenPort == NOT_ATTACHED) {
-            throw new RuntimeException("Failed to attach agent. On Jdk9+ you must run target process with -Djdk.attach.allowAttachSelf=true");
+            throw new RuntimeException("Failed to attach agent. On JDK 9 and higher, you must run JRD with -Djdk.attach.allowAttachSelf=true");
```

Comment:
This is wrong. You really have to run target process, not JRD:
   throw new RuntimeException("Failed to attach agent. On JDK 9 and higher, you must run target process with -Djdk.attach.allowAttachSelf=true");

It is the remte VM which have now protections agaisnt random agents. Only our super saimple case is that jrd checks itself

Reply:
Addressed in cedfc2c, sorry for the confusion on my part.

Reply:
yup. the only reason it is in jrd launcher, is so it can by default look into itself